azurerm/data_source_redis_cache.go

```diff
@@ -0,0 +1,98 @@
+package azurerm
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmRedisCache() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmRedisCacheRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
This isn't required on a datasource
```suggestion
```

Reply:
Pulling in the schema from the resource as suggested, will address this copy-paste faux pas as part of this.